src/app/components/Tokens/TokenHolders.tsx

```diff
@@ -0,0 +1,113 @@
+import { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Table, TableCellAlign, TableColProps } from '../Table'
+import { BareTokenHolder } from '../../../oasis-nexus/api'
+import { TablePaginationProps } from '../Table/TablePagination'
+import { AccountLink } from '../Account/AccountLink'
+import { fromBaseUnits } from '../../utils/helpers'
+import { TokenPriceInfo } from '../../../coin-gecko/api'
+
+type TableTokenHolder = BareTokenHolder & {
+  markAsNew?: boolean
+}
+
+type TokenHoldersProps = {
+  holders?: TableTokenHolder[]
+  isLoading: boolean
+  decimals: number | undefined
+  totalSupply: string | undefined
+  limit: number
+  tokenPrice?: TokenPriceInfo
+  pagination: false | TablePaginationProps
+}
+
+export const TokenHolders: FC<TokenHoldersProps> = ({
+  isLoading,
+  limit,
+  pagination,
+  holders,
+  decimals = 18,
+  totalSupply,
+  tokenPrice,
+}) => {
+  const { t } = useTranslation()
+  // const { isMobile } = useScreenSize()
+  const tableColumns: TableColProps[] = [
+    { key: 'rank', content: t('common.rank'), align: TableCellAlign.Center },
+    { key: 'address', content: t('common.address') },
+    { key: 'quantity', content: t('common.quantity'), align: TableCellAlign.Right },
+    { key: 'percentage', content: t('common.percentage'), align: TableCellAlign.Right },
+    ...(tokenPrice
+      ? [{ key: 'value', align: TableCellAlign.Right, content: t('common.value'), width: '250px' }]
+      : []),
+  ]
+
+  const calculateRatio = (balance: string): string => {
+    return totalSupply === undefined
+      ? t('common.missing')
+      : `${((100 * parseFloat(fromBaseUnits(balance, decimals))) / parseFloat(totalSupply)).toFixed(4)}%`
+  }
+
+  const tableRows = holders?.map((holder, index) => {
+    return {
+      key: `event-${index + (pagination ? (pagination.selectedPage - 1) * pagination.rowsPerPage : 0)}`,
+      data: [
+        {
+          key: 'rank',
+          content: (
+            index +
+            (pagination ? (pagination.selectedPage - 1) * pagination.rowsPerPage : 0) +
+            1
+          ).toLocaleString(),
+          align: TableCellAlign.Center,
+        },
+        {
+          key: 'address',
+          content: (
+            <AccountLink scope={holder} address={holder.eth_holder_address || holder.holder_address} />
+          ),
+        },
+        {
+          key: 'quantity',
+          content: t('tokens.totalSupplyValue', { value: fromBaseUnits(holder.balance, decimals) }),
```

Comment:
Shouldn't we use RoundedBalance here for example 
`http://localhost:1234/mainnet/sapphire/token/0x4344919960B0196c512807eE2f3922302fFFE542/holders#holders`

```
-- | --
1 | 0x3BA9F711C9808902f6a200d3715ba8a7e4D8b63f | 100 | 100.0000%
2 | 0x00000000000000000000000000000000000000ff | 0 | 0.0000%
```

we show 0 in the second row for example

Reply:
> Shouldn't we use RoundedBalance here for Quantity? 

OK done

Reply:
> Not sure about Percentage. for example `http://localhost:1234/mainnet/sapphire/token/0x4344919960B0196c512807eE2f3922302fFFE542/holders#holders`
> 
> ```
> -- | --
> 1 | 0x3BA9F711C9808902f6a200d3715ba8a7e4D8b63f | 100 | 100.0000%
> 2 | 0x00000000000000000000000000000000000000ff | 0 | 0.0000%
> ```
> 
> we show 0 in the second row

I don't see the issue here. The percentage must _obviously_ be a rounded value, so what else would anyone expect?

Reply:
OK, so I think we are good here.

Reply:
I would expect
![da222769 oasis-explorer pages dev_mainnet_sapphire_token_0x4344919960B0196c512807eE2f3922302fFFE542_holders (1)](https://github.com/oasisprotocol/explorer/assets/3758846/bd38f029-4352-4714-93f6-5e27da59a9de)
not
![da222769 oasis-explorer pages dev_mainnet_sapphire_token_0x4344919960B0196c512807eE2f3922302fFFE542_holders](https://github.com/oasisprotocol/explorer/assets/3758846/440bc39b-4791-4379-9281-b179bf7b7d65)